Exoft.Gamification.Api.Services/UserAchievementService.cs

```diff
@@ -78,10 +79,36 @@ public async Task DeleteAsync(Guid userAchievementId)
             _userAchievementRepository.Delete(userAchievement);
 
             user.Achievements.Remove(userAchievement);
+            user.XP -= userAchievement.Achievement.XP;
+
 
             await _unitOfWork.SaveChangesAsync();
         }
 
+        public async Task<AchievementsInfo> GetAchievementsInfoByUserAsync(Guid userId)
+        {
+            var countAchievements = await _userAchievementRepository.GetCountAchievementsByUserAsync(userId);
```

Comment:
Would be better to move some logic into repository, because here you retrieve all achievements, which is bad